### PR TITLE
add C++ exceptions (and Python) to C++ language bindings

### PR DESCRIPTION
**Pull request by mislavn**
This is the first version of the C++ exception refactoring.

@rkrejci please can you check if this patch works in your use case.

### CODE REVIEW

File: swig/cpp/src/Libyang.cpp [hunk 1]
```diff
@@ -36,27 +36,44 @@ Context::Context(ly_ctx *ctx, S_Deleter deleter):
...
 Context::~Context() {}
+int Context::set_searchdir(const char *search_dir) {
+    int ret = ly_ctx_set_searchdir(ctx, search_dir);
+    if (ret) {
+        check_libyang_error();
+    }
+    return ret;
+}
+void Context::unset_searchdirs(int idx) {
+    int ret = ly_ctx_unset_searchdirs(ctx, idx);
```
Comment (rkrejci) [inline swig/cpp/src/Libyang.cpp]: ly_ctx_unset_searchdirs() returns void, it does not fail.

**Comment by mislavn** [comment]
There is still some issues with:

```
>>> import libyang as ly
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: dynamic module does not define module export function (PyInit_libyang)
```

I'll take a look.

**Comment by mislavn** [comment]
I think I know what is the problem. I just need to adjust my cmakefiles for the commit fd7a5c72a7c45750166f1fa5b23d73250b4cf5fa .

**Comment by rkrejci** [comment]
So, should I wait with the merge? I'm not able to reproduce this issue when merged with `swig_change` branch.

**Comment by mislavn** [comment]
It's up to you. I'll probably fix the issue today.

**Pull request merged**